tsdb/wal/wal_test.go

```diff
@@ -484,3 +484,32 @@ func BenchmarkWAL_Log(b *testing.B) {
 		})
 	}
 }
+
+// TestCloseGoroutineLeak detect goroutine leak when call Close of wal instance.
+func TestCloseGoroutineLeak(t *testing.T) {
+	dir, err := ioutil.TempDir("", "wal_close_goroutine_leak")
+	testutil.Ok(t, err)
+	defer func() {
+		testutil.Ok(t, os.RemoveAll(dir))
+	}()
+	w, err := NewSize(nil, nil, dir, pageSize, false)
+	testutil.Ok(t, err)
+
+	closedSegment, err := CreateSegment(dir, 0)
+	testutil.Ok(t, err)
+	err = w.setSegment(closedSegment)
+	testutil.Ok(t, err)
```

Comment:
```suggestion
	testutil.Ok(t, w.setSegment(closedSegment))
```

Even for `w.Log()` and `closedSegment.Close()` below.